lib/selinuxtest.pm

```diff
@@ -93,4 +96,57 @@ sub check_category {
     }
 }
 
+sub set_sestatus {
+    my ($self, $mode, $type) = @_;
+    my $selinux_config_file = "/etc/selinux/config";
+    select_console "root-console";
+
+    # SELinux by default
+    validate_script_output("sestatus", sub { m/SELinux status: .*disabled/ });
+
+    # workaround for "selinux-auto-relabel" in case: auto relabel then trigger reboot
+    my $results = script_run("zypper --non-interactive se selinux-autorelabel");
+    if (!$results) {
+        assert_script_run("sed -ie \'s/GRUB_TIMEOUT.*/GRUB_TIMEOUT=8/\' /etc/default/grub");
+    }
+
+    # enable SELinux in grub
+    if ($mode eq "enforcing") {
+        add_grub_cmdline_settings('security=selinux selinux=1 enforcing=1', update_grub => 1);
+    }
+    elsif ($mode eq "permissive") {
+        add_grub_cmdline_settings('security=selinux selinux=1 enforcing=0', update_grub => 1);
+    }
+
+    # control (enable) the status of SELinux on the system, e.g., "enforcing" or "permissive"
+    assert_script_run("sed -i -e 's/^SELINUX=/#SELINUX=/' $selinux_config_file");
+    assert_script_run("echo SELINUX=$mode >> $selinux_config_file");
+
+    # set SELINUXTYPE, e.g., "minimum" or "targeted"
+    assert_script_run("sed -i -e 's/^SELINUXTYPE=/#SELINUXTYPE=/' $selinux_config_file");
+    assert_script_run("echo SELINUXTYPE=$type >> $selinux_config_file");
+    assert_script_run("systemctl enable auditd");
+
+    # reboot the vm and reconnect the console
+    power_action("reboot", textmode => 1);
+    reconnect_mgmt_console if is_pvm;
```

Comment:
I suggest that your test module avoid handling any backend specific behaviour. Better if abstracted lib functions ensure this.  Your comment in line L130 already is a good name for a function that you could implement, e.g.

```perl
sub reboot_and_reconnect {
    my (%args) = @_;
    power_action("reboot", textmode => $args{textmode});
    reconnect_mgmt_console if is_pvm;
    $self->wait_boot(textmode => $args{textmode});
}
```

Reply:
Done. Thanks.